core/optaplanner-constraint-streams-bavet/src/main/java/org/optaplanner/constraint/streams/bavet/common/AbstractJoinNode.java

```diff
@@ -248,16 +234,15 @@ private void retractTuple(OutTuple_ outTuple) {
                 // Kill it before it propagates
                 outTuple.setState(BavetTupleState.ABORTING);
                 break;
+            case OK:
+                dirtyTupleQueue.add(outTuple);
+                // Intentional fall-through.
             case UPDATING:
                 // Don't add the tuple to the dirtyTupleQueue twice
                 // Kill the original propagation
                 outTuple.setState(BavetTupleState.DYING);
                 break;
             // DYING and ABORTING are impossible because they shouldn't linger in the indexes.
-            case OK:
-                outTuple.setState(BavetTupleState.DYING);
-                dirtyTupleQueue.add(outTuple);
-                break;
```

Comment:
IMO, it's clearer if they're separate and in order than as few lines as possible. Perf wise it won't make a difference. Wdyt?